From 7eb35b2a9180c32f03da64e8c3453f59a2822cd9 Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Wed, 30 Nov 2022 11:38:17 -0800
Subject: [PATCH] broadmobi: add MM_PLUGIN_REQUIRED_QCDM flag

This fixes an issue where, with an BM818 connected over USB,
suspending/resuming the modem causes the modem to reset immediately.

The culprit was bisected to d6203265c2, where QCDM probing was disabled
unless explicitly requested, so I assume this modem needs it based on
the constant resetting when power management was enabled on the modem's
usb connection.
---
 plugins/broadmobi/mm-plugin-broadmobi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/broadmobi/mm-plugin-broadmobi.c b/plugins/broadmobi/mm-plugin-broadmobi.c
index 0c91f728..805663fe 100644
--- a/plugins/broadmobi/mm-plugin-broadmobi.c
+++ b/plugins/broadmobi/mm-plugin-broadmobi.c
@@ -76,7 +76,7 @@ mm_plugin_create (void)
                       MM_PLUGIN_ALLOWED_SUBSYSTEMS, subsystems,
                       MM_PLUGIN_ALLOWED_VENDOR_IDS, vendor_ids,
                       MM_PLUGIN_ALLOWED_AT,         TRUE,
-                      MM_PLUGIN_ALLOWED_QCDM,       TRUE,
+                      MM_PLUGIN_REQUIRED_QCDM,      TRUE,
                       MM_PLUGIN_ALLOWED_QMI,        TRUE,
                       NULL));
 }
-- 
2.38.1

